src/tls.rs

```diff
@@ -0,0 +1,138 @@
+use std::io::{BufReader, Read, Write};
+use std::sync::{Arc, Mutex};
+use std::{fs, io};
+
+use rustls::{self, Connection, NoClientAuth, ServerConfig, ServerConnection};
+
+/// TLS configuration
+#[derive(Clone)]
+pub struct TlsConfig {
+    /// Full path to the server certificate file.
+    pub server_cert: String,
+    /// Full path to the server key file.
+    pub server_cert_key: String,
+}
+
+fn make_config(config: &TlsConfig) -> Result<Arc<rustls::ServerConfig>, io::Error> {
+    let builder = ServerConfig::builder()
```

Comment:
It seems unfortunate for us to hard-code these things here. Can we instead let the user provide us with a `rustls::ServerConfig` directly?

Reply:
Yes. I think I was trying to avoid exposing the fact that we're using rustls, but this is worse.  I'll change this.